src/js/components/LogRow.js

```diff
@@ -19,51 +20,69 @@ type Props = {
+  onDoubleClick: () => void,
   rightClick: RightClickBuilder
 }
 
-export default class LogRow extends React.Component<Props> {
-  shouldComponentUpdate(nextProps: Props) {
-    return (
-      !Log.isSame(this.props.log, nextProps.log) ||
-      !isEqual(this.props.columns, nextProps.columns) ||
-      this.props.highlight !== nextProps.highlight ||
-      this.props.dimens.rowWidth !== nextProps.dimens.rowWidth ||
-      this.props.timeZone !== nextProps.timeZone ||
-      this.props.timeFormat !== nextProps.timeFormat
-    )
-  }
+const LogRow = (props: Props) => {
+  const {
+    dimens,
+    highlight,
+    index,
+    log,
+    rightClick,
+    columns,
+    onClick,
+    onDoubleClick
+  } = props
 
-  render() {
-    const {dimens, highlight, index, log, rightClick, columns} = this.props
-    const renderCell = (column, colIndex) => {
-      const width = dimens.rowWidth !== "auto" ? column.width || 300 : "auto"
-      const field = log.field(column.name)
-      const key = `${index}-${colIndex}`
-      if (field) {
-        return (
-          <LogCell
-            rightClick={rightClick}
-            key={key}
-            field={field}
-            log={log}
-            style={{width}}
-          />
-        )
-      }
-      if (dimens.rowWidth !== "auto") {
-        return <div className="log-cell" key={key} style={{width}} />
-      }
-    }
+  const handleClick = useDoubleClick(onClick, onDoubleClick)
 
-    return (
-      <div
-        className={classNames("log-row", {highlight, even: index % 2 == 0})}
-        style={Styler.row(dimens)}
-        onClick={this.props.onClick}
-      >
-        {columns.getVisible().map(renderCell)}
-      </div>
-    )
+  const renderCell = (column, colIndex) => {
+    const width = dimens.rowWidth !== "auto" ? column.width || 300 : "auto"
+    const field = log.field(column.name)
+    const key = `${index}-${colIndex}`
+    if (field) {
+      return (
+        <LogCell
+          rightClick={rightClick}
+          key={key}
+          field={field}
+          log={log}
+          style={{width}}
+        />
+      )
+    }
+    if (dimens.rowWidth !== "auto") {
+      return (
+        <div
+          className="log-cell"
+          key={key}
+          style={{width}}
+          onClick={handleClick}
+        />
+      )
+    }
   }
+
+  return (
+    <div
+      className={classNames("log-row", {highlight, even: index % 2 === 0})}
+      style={Styler.row(dimens)}
+      onClick={handleClick}
+    >
+      {columns.getVisible().map(renderCell)}
+    </div>
+  )
 }
+
+export default memo<Props>(LogRow, (prevProps: Props, nextProps: Props) => {
+  return !(
+    !Log.isSame(prevProps.log, nextProps.log) ||
+    !isEqual(prevProps.columns, nextProps.columns) ||
+    prevProps.highlight !== nextProps.highlight ||
+    prevProps.dimens.rowWidth !== nextProps.dimens.rowWidth ||
+    prevProps.timeZone !== nextProps.timeZone ||
+    prevProps.timeFormat !== nextProps.timeFormat
+  )
```

Comment:
How about inverting this to avoid the double negatives?

```js
return Log.isSame(prevProps.log, nextProps.log) &&
  isEqual(prevProps.columns, nextProps.columns) &&
  prevProps.highlight === nextProps.highlight &&
  prevProps.dimens.rowWidth === nextProps.dimens.rowWidth &&
  prevProps.timeZone === nextProps.timeZone &&
  prevProps.timeFormat === nextProps.timeFormat
```

Reply:
I'm glad the inverted the API, I think it makes more sense to say, "use the memo'ed version when this condition is true."

Reply:
I had the same thought actually. Glad you agree, will change